src/examples/spi/spi.go

```diff
@@ -0,0 +1,29 @@
+// Example using the spi hardware interface on the Arduino
+//
+package main
+
+import (
+	"machine"
+	"time"
+)
+
+func main() {
+	cs := machine.SPI0_CS_PIN
+	cs.High()
+	cs.Configure(machine.PinConfig{Mode: machine.PinOutput})
```

Comment:
@deadprogram do you have an opinion on whether to include an example here?

In any case, if you include an example you should also add a line to the smoke tests in the Makefile to make sure it continues to compile.

Reply:
Happy to add to the smoke tests.  I apologize for the broken build & missing this.  I tried running the tests, but since I didn't rebuild llvm from scratch something in the make file doesn't seem happy.  I'm working on running them one by one.

Reply:
You don't need to build with a custom LLVM build. At least if you do `make smoketest TINYGO=tinygo` you can pick the binary that you'd like to use.

Reply:
Seems like there will be no need for a specific example, by standardizing a bit more with the other SPI implementations. Specifically, removing  `[]CSPins`.